users/test_update_authorization.py

```diff
@@ -0,0 +1,112 @@
+        self.assertTrue(self.mock_collection.find_one(
+            {"user_id": AUTHORIZED_USER})["is_organizer"])
+        app.update_user_authorization_in_db(
+            AUTHORIZED_USER, False, self.mock_collection)
+        self.assertFalse(self.mock_collection.find_one(
+            {"user_id": AUTHORIZED_USER})["is_organizer"])
+        # False to True
+        self.assertFalse(self.mock_collection.find_one(
+            {"user_id": NON_AUTHORIZED_USER})["is_organizer"])
+        app.update_user_authorization_in_db(
+            NON_AUTHORIZED_USER, True, self.mock_collection)
+        self.assertTrue(self.mock_collection.find_one(
+            {"user_id": NON_AUTHORIZED_USER})["is_organizer"])
+
+    def test_keep_authorization_the_same(self):
+        """Can trivially change the authorization to the same value."""
+        # True to True
+        self.assertTrue(self.mock_collection.find_one(
+            {"user_id": AUTHORIZED_USER})["is_organizer"])
+        app.update_user_authorization_in_db(
+            AUTHORIZED_USER, True, self.mock_collection)
+        self.assertTrue(self.mock_collection.find_one(
+            {"user_id": AUTHORIZED_USER})["is_organizer"])
+        # False to False
+        self.assertFalse(self.mock_collection.find_one(
+            {"user_id": NON_AUTHORIZED_USER})["is_organizer"])
+        app.update_user_authorization_in_db(
+            NON_AUTHORIZED_USER, False, self.mock_collection)
+        self.assertFalse(self.mock_collection.find_one(
+            {"user_id": NON_AUTHORIZED_USER})["is_organizer"])
+
+    def test_set_authorization_of_malformed_user(self):
+        """Can set the authorization of a user missing the attribute in db."""
+        self.assertNotIn("is_organizer", self.mock_collection.find_one(
+            {"user_id": MALFORMATTED_IN_DB_USER}))
+        app.update_user_authorization_in_db(
+            MALFORMATTED_IN_DB_USER, False, self.mock_collection)
+        self.assertFalse(self.mock_collection.find_one(
+            {"user_id": MALFORMATTED_IN_DB_USER})["is_organizer"])
+
+    def test_fail_on_missing_user(self):
+        """Should raise error and do nothing on user missing from db."""
+        self.assertIsNone(self.mock_collection.find_one(
+            {"user_id": MISSING_USER}))
+        with self.assertRaises(KeyError):
+            app.update_user_authorization_in_db(
+                MISSING_USER, False, self.mock_collection)
+        # make sure no user was inserted
+        self.assertIsNone(self.mock_collection.find_one(
+            {"user_id": MISSING_USER}))
+
+    def test_fail_on_bad_arg_type(self):
+        """Should raise error and do nothing on bad arg type."""
+        self.assertTrue(self.mock_collection.find_one(
+            {"user_id": AUTHORIZED_USER})["is_organizer"])
+        with self.assertRaises(TypeError):
+            app.update_user_authorization_in_db(
+                AUTHORIZED_USER, "I am not a bool", self.mock_collection)
```

Comment:
`"False"` might be a good value to test with here, because in python, `"False"` is a non-empty string, so `bool("False")` is `True`.

Using `"False"` as an example also suggests *why* you care about this case.

Reply:
So in app.update_user_authorization_in_db should I still raise an error when the type is not bool like before, or should duck typing with `bool("False")` is `True` be the expected behaviour?

Reply:
I think the error raising is still necessary - with this approach the code may behave confusingly (for example if is_organizer is a string) and it seems best to make sure that can't happen altogether

Reply:
Done https://github.com/knative-portability/large-events/pull/11/commits/3f19cef83509aca7fa05b0aa9f06f2c1f783d186